docs/python 2 sunset.txt

```diff
@@ -0,0 +1,35 @@
+:orphan:
+
+Python 2 Sunset
+===============
+
+Since January 2020 and the release of Setuptools 45, Python 2 is no longer supported by the most current release (`discussion <https://github.com/pypa/setuptools/issues/1458>`_). Setuptools as a project continues to support Python 2 with bugfixes and important features on Setuptools 44.x.
+
+By design, most users will be unaffected by this change. That's because Setuptools 45 declares its supported Python versions to exclude Python 2.7, and installers such as pip 9 or later will honor this declaration and prevent installation of Setuptools 45 or later in Python 2 environments.
+
+Users that do import any portion of Setuptools 45 or later on Python 2 are directed to this documentation to provide guidance on how to work around the issues.
+
+Workarounds
+-----------
+
+The best recommendation is to avoid Python 2 and move to Python 3 where possible. This project acknowledges that not all environments can drop Python 2 support, so provides other options.
+
+In less common scenarios, later versions of Setuptools can be installed on unsupported Python versions. In these environments, the installer is advised to first install ``setuptools<45`` to "pin Setuptools" to a compatible version.
+
+- When using older versions of pip (before 9.0), the ``Requires-Python`` directive is not honored and invalid versions can be installed. Users are advised first to upgrade pip and retry or to pin Setuptools. Use ``pip --version`` to determine the version of pip.
+- When using ``easy_install``, ``Requires-Python`` is not honored and later versions can be installed. In this case, users are advised to pin Setuptools. This applies to ``setup.py install`` invocations as well, as they use Setuptools under the hood.
+
+It's still not working
+----------------------
+
+If after trying the above steps, the Python environment still has incompatible versions of Setuptools installed, here are some things to try.
+
+1. Uninstall and reinstall Setuptools. Run ``pip uninstall -y setuptools`` for the relevant environment. Repeat until there is no Setuptools installed. Then ``pip install setuptools``.
+
+2. If possible, attempt to replicate the problem in a second environment (virtual machine, friend's computer, etc). If the issue is isolated to just one unique enviornment, first determine what is different about those environments (or reinstall/reset the failing one to defaults).
+
+3. End users who are not themselves the maintainers for the package they are trying to install should contact the support channels for the relevant application. Please be respectful by searching for existing issues and following the latest guidance before reaching out for support. When filing an issue, be sure to give as much detail as possible to help the maintainers understand what factors led to the issue after following their recommended guidance.
```

Comment:
English is not my native language, but *please be respectful* somehow sounds a bit odd here.  In addition, I think enumerate list elements should not be separated, i.e.
```rst
1. Uninstall and reinstall Setuptools...
2. If possible, attempt to replicate the problem...
```

Reply:
The reason I say "Please be respectful" is I'm directing users at other maintainers, and I want to be cautious not to create a deluge of support requests to other projects. Perhaps it would help to say "Please be respectful to those projects" or "Please be considerate".

> list elements should not be separated

I separated them because it makes it easier for me to parse visually when in text form. I don't think it makes a difference for rendering. If it's just a matter of opinion, I'll stick with what's there. If there's guidance that not having the intervening lines is better, I'm happy to align with best practices.

Reply:
I think giving more context (either of *Please be respectful/considerate to those projects*) might be better.

If the markup language is reST (I took a look at `conf.py` and saw no language set, so I think it's the default), I'll remove the lines in the follow-up PR since it will not look bad in plain text, e.g.
```rst
1. One two three
   four five six
2. Un deux trois
   quart cinq six

Reply:
Indeed, that looks great. I'll incorporate it.

Reply:
Done in dbdd19d4 and f7b7ee25.